### PR TITLE
ci: add release workflow without pypa action

### LINKED ISSUE #68: No publish to PyPI workflow setup
There is no publish workflow setup for this project yet.

This is what we should add I think: https://github.com/jupyterhub/jupyterhub-idle-culler/blob/master/.github/workflows/publish.yml

@minrk @yuvipanda could you add jupyterhub-bot to https://pypi.org/project/sudospawner/? Then I'll setup a deployment token for this project and a GitHub secret named `PYPI_PASSWORD` like used in the workflow referenced above.

### PR DESCRIPTION
**Pull request by consideRatio**
Closes #68 by adding a release workflow. Note that I'm currently suggesting that we stop using the PyPA action as it is pretty much a one-liner to do it manually.

What is done by the action is can be seen in the PyPA action can be seen in [this script](https://github.com/pypa/gh-action-pypi-publish/blob/master/twine-upload.sh) based on [these default input variables](https://github.com/pypa/gh-action-pypi-publish/blob/master/action.yml) using `twine` with the following `--help` message for the check and upload commands.

```
usage: twine upload [-h] [-r REPOSITORY] [--repository-url REPOSITORY_URL]
                    [-s] [--sign-with SIGN_WITH] [-i IDENTITY] [-u USERNAME]
                    [-p PASSWORD] [--non-interactive] [-c COMMENT]
                    [--config-file CONFIG_FILE] [--skip-existing]
                    [--cert path] [--client-cert path] [--verbose]
                    [--disable-progress-bar]
                    dist [dist ...]

positional arguments:
  dist                  The distribution files to upload to the repository
                        (package index). Usually dist/* . May additionally
                        contain a .asc file to include an existing signature
                        with the file upload.

optional arguments:
  -h, --help            show this help message and exit
  -r REPOSITORY, --repository REPOSITORY
                        The repository (package index) to upload the package
                        to. Should be a section in the config file (default:
                        pypi). (Can also be set via TWINE_REPOSITORY
                        environment variable.)
  --repository-url REPOSITORY_URL
                        The repository (package index) URL to upload the
                        package to. This overrides --repository. (Can also be
                        set via TWINE_REPOSITORY_URL environment variable.)
  -s, --sign            Sign files to upload using GPG.
  --sign-with SIGN_WITH
                        GPG program used to sign uploads (default: gpg).
  -i IDENTITY, --identity IDENTITY
                        GPG identity used to sign files.
  -u USERNAME, --username USERNAME
                        The username to authenticate to the repository
                        (package index) as. (Can also be set via
                        TWINE_USERNAME environment variable.)
  -p PASSWORD, --password PASSWORD
                        The password to authenticate to the repository
                        (package index) with. (Can also be set via
                        TWINE_PASSWORD environment variable.)
  --non-interactive     Do not interactively prompt for username/password if
                        the required credentials are missing. (Can also be set
                        via TWINE_NON_INTERACTIVE environment variable.)
  -c COMMENT, --comment COMMENT
                        The comment to include with the distribution file.
  --config-file CONFIG_FILE
                        The .pypirc config file to use.
  --skip-existing       Continue uploading files if one already exists. (Only
                        valid when uploading to PyPI. Other implementations
                        may not support this.)
  --cert path           Path to alternate CA bundle (can also be set via
                        TWINE_CERT environment variable).
  --client-cert path    Path to SSL client certificate, a single file
                        containing the private key and the certificate in PEM
                        format.
  --verbose             Show verbose output.
  --disable-progress-bar
                        Disable the progress bar.
```

```
usage: twine check [-h] dist [dist ...]

positional arguments:
  dist        The distribution files to check, usually dist/*

optional arguments:
  -h, --help  show this help message and exit
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        run: |
+          python -m build --sdist --wheel .
+          ls -l dist
+
+      - name: verify release metadata
+        run: python -m twine check dist/*
+
+      - name: upload release to pypi
+        if: startsWith(github.ref, 'refs/tags/')
+        run: twine upload --username __token__ --password ${{ secrets.pypi_password }} dist/*
```
Comment (minrk) [inline .github/workflows/release.yml:36]: Maybe use TWINE_USERNAME, TWINE_PASSWORD [environment variables](https://twine.readthedocs.io/en/latest/#environment-variables)? Even though GHA logs hide them, I still feel a bit better passing credentials via env than command-line.

Comment (consideRatio) [inline .github/workflows/release.yml:36]:
```suggestion
        env:
          TWINE_USERNAME: __token__
          TWINE_PASSWORD: ${{ secrets.pypi_password }}
        run: twine upload dist/*
```

**Comment by consideRatio** [comment]
@minrk, this has gone stale - does it LGTY?

**Comment by minrk** [comment]
Thanks for the reminder!

**Pull request merged**